src/utils/currencies.js

```diff
@@ -40,7 +40,7 @@ function getFixedTokenAmount(amount, decimals) {
   // We can lose precision as there will never be more than MAX_DECIMALS_UNTIL_ZERO_AMOUNT significant digits
   const balanceWithDecimals = Number(amount) / Math.pow(10, decimals);
 
-  return Number(balanceWithDecimals.toFixed(MAX_TOKEN_DECIMALS)).toString();
+  return balanceWithDecimals.toFixed(MAX_TOKEN_DECIMALS);
```

Comment:
Why change this? I think adding Number() helps remove prefixed 0s

Reply:
People may put a leading 0 in the input form. If you convert to number, it makes that impossible

Reply:
But `balanceWithDecimals` is a number and toFixed return a string without leading zeros and `Number(amount)` should return previously a number without leading zeros

Reply:
Understood!